mergify_engine/web.py

```diff
@@ -151,6 +157,70 @@ def subscription_cache(installation_id):  # pragma: no cover
     return "Cache cleaned", 200
 
 
+class PullRequestUrlInvalid(voluptuous.Invalid):
+    pass
+
+
+@voluptuous.message('expected a Pull Request URL', cls=PullRequestUrlInvalid)
+def PullRequestUrl(v):
+    _, owner, repo, _, pull_number = urlsplit(v).path.split("/")
+    pull_number = int(pull_number)
+
+    integration = github.GithubIntegration(config.INTEGRATION_ID,
+                                           config.PRIVATE_KEY)
+    installation_id = utils.get_installation_id(integration, owner)
+    if not installation_id:  # pragma: no cover
+        raise PullRequestUrlInvalid(
+            message="Mergify not installed on this repository"
+        )
+
+    token = integration.get_access_token(installation_id).token
+    try:
+        return mergify_pull.MergifyPull.from_number(
+            installation_id, token, owner, repo, pull_number
+        )
+    except github.UnknownObjectException:
+        raise PullRequestUrlInvalid(
+            message="Pull request not found"
+        )
+
+
+SimulatorSchema = voluptuous.Schema({
+    voluptuous.Required('pull_request'): PullRequestUrl(),
+    voluptuous.Required('mergify.yml'): rules.UserConfigurationSchema,
+})
+
+
+@app.errorhandler(voluptuous.Invalid)
+def voluptuous_error(error):
+    payload = flask.jsonify({
+        "message": str(error),
+        "type": error.__class__.__name__,
+        "error": error.msg,
```

Comment:
Is this really needed? sounds like `message` is enough

Reply:
That's not the same information: https://github.com/alecthomas/voluptuous/blob/master/voluptuous/error.py#L27-L36